hist/hist/src/Haxis.cxx

```diff
@@ -46,6 +46,55 @@ Color_t TH1::GetAxisColor( Option_t *axis) const
    return 0;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Return the histogram bar offset along X or Y axis.
+///
+/// axis specifies which axis ("x","y"), default = "x"
+///
+///  - if axis="x"  The bar offset along the X axis is returned
+///  - if axis="y"  The bar offset along the Y axis is returned
+
+Float_t  TH1::GetBarOffset( Option_t *axis) const
+{
+   if (fBarOffset <= 1000) return Float_t(0.001*Float_t(fBarOffset));
+
+   Int_t ax = AxisChoice(axis);
+   if (ax == 1) {
+      if (fBarOffset <= 2000 ) return Float_t(0.001*Float_t(fBarOffset-1000));
+      if (fBarOffset <= 3000 ) return 0.;
+   }
+   if (ax == 2) {
+      if (fBarOffset <= 2000 ) return 0.;
+      if (fBarOffset <= 3000 ) return Float_t(0.001*Float_t(fBarOffset-2000));
+   }
+
+   return 0.;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return the histogram bar width along X or Y axis.
+///
+/// axis specifies which axis ("x","y"), default = "x"
+///
+///  - if axis="x"  The bar width along the X axis is returned
+///  - if axis="y"  The bar width along the Y axis is returned
+
+Float_t TH1::GetBarWidth( Option_t *axis) const
+{
+   if (fBarWidth <= 1000) return Float_t(0.001*Float_t(fBarWidth));
+
+   Int_t ax = AxisChoice(axis);
+   if (ax == 1) {
+      if (fBarWidth <= 2000 ) return Float_t(0.001*Float_t(fBarWidth-1000));
```

Comment:
Here should be:

     if (fBarWidth <= 2001 ) return Float_t(0.001*Float_t(fBarWidth-1001));